### PR TITLE
NgRx Component Store -> StateAdapt

### PR DESCRIPTION
**Pull request by mfp22**
At 378.13 KB it isn't as small as NgRx Component Store, but I think it's the smallest that supports Redux Devtools so far. 

Starting from NgRx Component Store I added 179 loc and deleted 221 loc.

### CODE REVIEW

File: src/app/products/product-shell/product-shell.component.ts [hunk 1]
```diff
@@ -1,46 +1,9 @@
...
-import {
-    ProductStateFacadeService,
-} from '../state/product-state-facade.service';
+import { Component } from '@angular/core';
+import { ProductStoreService } from '../state/product.store.service';
 
 @Component({
     templateUrl: './product-shell.component.html',
 })
-export class ProductShellComponent implements OnInit {
```
Comment (spierala) [inline src/app/products/product-shell/product-shell.component.ts:10]: @mfp22 Thanks for the PR :)

The changes in the component code should be kept to a minimum.

All other refactors still have this code.

The main refactor should happen in de facade to make the comparison as fair as possible. That way the solutions can also be compared more easily.

File: src/app/user/login.component.ts [hunk 2]
```diff
@@ -23,10 +23,6 @@ export class LoginComponent implements OnInit {
         this.router.navigate(['welcome']);
     }
 
-    checkChanged(): void {
```
Comment (spierala) [inline src/app/user/login.component.ts:26]: Also here, please keep the component code as much as possible.

File: src/app/products/product-shell/product-shell.component.ts (hunk 1 above)
Comment (mfp22) [reply src/app/products/product-shell/product-shell.component.ts:10]: Methods are containers for imperative code, none of which should exist between component outputs and StateAdapt sources (or even just purely reactive RxJS code). However, for sake of comparison, I will add the methods back. But I consider the 100% reactive architecture recommendation an important aspect of StateAdapt.

**Comment by spierala** [comment]
Closing this PR. I did not see any activity for some time :)

FYI I updated the contribution guidelines: https://github.com/spierala/angular-state-management-comparison/blob/master/README.md#contributing

I would love to see StateAdapt 2.0 in the list.

Feel free to refactor without facade. But we have to mention it in the comment in the README.

**Pull request closed without merging**